tests/testthat/test-map.R

```diff
@@ -0,0 +1,185 @@
+
+test_that("vec_map() handles S3 vectors", {
+  vctr <- new_vctr(1:3)
+  rcrd <- new_rcrd(list(x = 1:3, y = 4:6))
+
+  expect_identical(
+    vec_map(vctr, identity),
+    vec_chop(vctr)
```

Comment:
I think it would be better to define the expected output directly rather than in terms of another function, because I don't have enough of `vec_chop()` loaded in my head to have any idea what this does.